File: go/Driver.go
package main

import (
	"fmt"

	structs "./structs"
)

func testLinkedList() {
	var ll *structs.LinkedList = new(structs.LinkedList)
	ll.Add("Hello")
	ll.Add("World")
	fmt.Println(ll.Size())
	if ll.Contains("World") {
		fmt.Println("World inside of the Linked List!")
	} else {
		fmt.Println("World not inside of the Linked List")
	}
	g, e := ll.Get(1)
	if e != nil {
		fmt.Println(e)
	} else {
		fmt.Println(g)
	}
	r, e2 := ll.Remove(1)
	if e2 != nil {
		fmt.Println(e2)
	} else {
		fmt.Println(r)
	}

	fmt.Println(ll.Size())
}

func testSet() {
	var s *structs.Set = new(structs.Set)
	s.Add(1)
	s.Add(2)
	s.Add(1)
	s.Add(1)
	s.Add(3)

	s.Print()
}

func testQueue() {
	var q *structs.Queue = new(structs.Queue)
	q.Enqueue(1)
	q.Enqueue(2)

	q.Print()

	fmt.Println()

	q.Dequeue()

	q.Print()
}

func testStack() {
	var s *structs.Stack = new(structs.Stack)
	s.Push(1)
	s.Push(2)
	s.Push(3)
	s.Print()

	fmt.Println()

	fmt.Printf("Popped: %d\n", s.Pop())

	s.Print()

}

func main() {
	// testLinkedList()
	// testSet()
	// testQueue()
	testStack()
}

File: php/Driver.php
<?php
include "structs/Set.php";
include "structs/ArrayList.php";
include "structs/LinkedList.php";
include "structs/Queue.php";
include "structs/PriorityQueue.php";
include "algorithms/Search.php";
include "algorithms/Factorial.php";

class TestLinkedList
{
    public static function main()
    {
        $ll = new LinkedList();
        $ll -> add("Hello");
        $ll -> add("World");
        $ll -> add("This");
        $ll -> add("Is");
        $ll -> add("A");
        $ll -> add("Test");

        echo "Stored Values: ";
        echo $ll;
        echo "\n";
    }
}

class TestQueue
{
    public static function main()
    {
        $queue = new Queue();
        $queue -> enqueue("Hello");
        $queue -> enqueue("World");
    
        echo "Enqueued: ";
        echo $queue;
        echo "\n";
    
        echo "Dequeued: ";
        echo $queue -> dequeue();
        echo "\n";
    
    
        echo $queue;
    }
}

class TestSet
{
    public static function main()
    {
        $set = new Set();
        $set -> add("Hello");
        $set -> add("Hello");
        $set -> add("Bye");
    
        echo $set -> size();
    }
}

class TestPriorityQueue
{
    public static function main()
    {
        $set = new Set();
        $set -> add("Hello");
        $set -> add("Hello");
        $set -> add("World");
  
        $ll = new LinkedList();
        $ll -> add("Hello");
        $ll -> add("World");
        $ll -> add("Test");

        $p_queue = new PriorityQueue();
        $p_queue -> insert("Hello");
        $p_queue -> insert("John", 20);
        $p_queue -> insert("World");
        $p_queue -> insert("Edward", 100);
        $p_queue -> insert($set);
        $p_queue -> insert($ll, 30);

        $p_queue -> insert("top", 50);
    
        echo $p_queue . "\n\n";

        echo "Deleted: \n" . $p_queue -> delete() . "\n\n";

        $p_queue -> insertAtTop("Top Value");
        $p_queue -> insertAtBottom("Bottom Value");
        $p_queue -> insertAtBottom("True bottom");

        $p_queue -> insert("Hello again");
    
        echo $p_queue;
        echo "\n\n";
        $p_queue -> resetCount();

        $p_queue -> insertAtTop("Top Value 2");
        $p_queue -> insert("Second top");
        $p_queue -> insertAtBottom("New Bottom");
        echo $p_queue;

        $p_queue -> resetCount();

        echo "\n\nDeleted: " . $p_queue -> delete();

        echo "\n\n";
        echo $p_queue;

        
    }
}

TestPriorityQueue::main();


class TestAlgorithms{
  public static function jumpSearchTest($collection){
    echo Search::jumpSearch($collection, 2);
  }

  public static function factorialTest(){
    echo "Recursive: 10! = " . Factorial::recursive(10) . "\n";
    echo "Iterative: 10! = " . Factorial::iterative(10) . "\n";
  }

  public static function main(){
    $collection = array(1,1,2,3,5,8,13,21,34);
    jumpSearchTest($collection);
  }
}

// TestAlgorithms::main();
